Add `SumOfSlatersPrep2` to `labs`. (#9539)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Labs: a place for unified and rapid prototyping of research software 🧪</h3>

* Added a variant of `SumOfSlatersPrep` to labs, accessible as `labs.templates.SumOfSlatersPrep2`.
  This variant handles work wires explicitly instead of allocating them dynamically in the

>   decomposition. This makes it possible to use `SumOfSlatersPrep2` with `qp.qjit` with capture disabled,

Reviewer: ```suggestion
  decomposition. This makes it possible to use `SumOfSlatersPrep2` with `qp.qjit` with capture disabled.
```

>   decomposition. This makes it possible to use `SumOfSlatersPrep2` with `qp.qjit` with capture disabled,

Reviewer 2: ```suggestion
  decomposition. This enables usage of `SumOfSlatersPrep2` with `qp.qjit` with capture _disabled_ (`qp.capture.disable()`).
```
